lib/db_connection.ex

```diff
@@ -378,145 +378,14 @@ defmodule DBConnection do
   """
   @callback disconnect(err :: Exception.t, state :: any) :: :ok
 
+  @optional_callbacks handle_first: 4, handle_next: 4, handle_info: 2, ping: 1
```

Comment:
* handle_first/next should be removed as they have been replaced by handle_fetch in 2.0, keeping those after a major version bump would be confusing and prevent us simplifying the code
* handle_info wont be called with the new pool so we should remove it
* ping should become required as we no longer have active mode with the new pool, so rely on ping to detect network drops

Reply:
I have done removed handle_first and handle_next. I have also made `ping` required but I am not sure about `handle_info/2`. The TCP example you have in examples defines `handle_info/2`... if that is not going to be called, how can users implement that part of the code?

Reply:
They can't anymore.